content/chapter-sources/webhook/_index.md

```diff
@@ -0,0 +1,161 @@
+
+Example minimal config:
+
+```shell
+source s_webhook {
+webhook-json(
+    port(8181)
+    paths(["/events","/events/(?P<HOST>.*)"])
+    );
+};
+```
+
+To test the source, you can use `curl` for example, on the host where {{< product >}} is running:
+
+```shell
+curl -X POST --data "{'MESSAGE':'message-value'}" http://127.0.0.1:8181/events
+```
+
+This driver is actually a reusable configuration snippet based on a [custom Python source]({{< relref "/chapter-sources/python-source/_index.md" >}}). For details on using or writing such configuration snippets, see {{% xref "/chapter-configuration-file/large-configs/config-blocks/_index.md" %}}. You can find the source of this configuration snippet on [GitHub](https://github.com/axoflow/axosyslog/blob/main/modules/python-modules/syslogng/modules/webhook/scl/webhook.conf).
+
+## HTTPS webhook
+
+To receive data using HTTPS, configure a certificate and a private key for the webhook using the `tls_cert_file` and `tls_key_file` options.
+
+To verify the certificate of the clients sending data to the webhook, set `tls_peer_verify(yes)`, and use the following options:
+
+- `tls_use_system_cert_store(no)`, or
+- `tls_ca_file("")`, or
+- `tls_ca_dir("")`
+
+## Options
+
+## auth_token()
+
+<!-- FIXME -->
+
+## include_request_headers()
+
+|          |         |
+| -------- | ------- |
+| Type:    | `yes` or `no` |
+| Default: |   `no`   |
+
+Available in version 4.11 and later.
+
+*Description:* If enabled, the HTTP request headers from the webhook will available for processing. For example, when using the `webhook-json()` source, the headers will added as a JSON object to the `${&lt;prefix>.headers}` field.
+
+## paths()
+
+|          |         |
+| -------- | ------- |
+| Type:    | JSON list |
+| Default: |      |
+
+*Description:* Sets the endpoints where the webhook will receive data. You can use static paths, or regular expressions. In regular expressions you can use named capture groups to automatically set macro values.
+
+For example, the `/events/(?P<HOST>.*)` path sets the hostname for the data received in the request based on the second part of the URL: a request to the `/events/my-example-host` URL sets the host field of that message to `my-example-host`.
+
+You can set multiple endpoints, for example, `paths(["/events","/events/(?P<HOST>.*)"])`
+
+## port()
+
+|          |         |
+| -------- | ------- |
+| Type:    | integer |
+| Default: |      |
+
+The port number where the webhook is listening on, for example, `8080`
+
+## prefix()
+
+|           |           |
+| --------- | --------- |
+| Type: | string  |
+| Default:  |  |
+
+*Description:* Insert a prefix before the name part of the parsed name-value pairs to help further processing when using the `webhook-json()` source. For example, to insert the `webhook.` prefix, use the `prefix(webhook.)` option.
+
+{{< include-headless "chunk/p-parser-prefix.md" >}}
+
+## proxy_header()
+
+Available in version 4.11 and later.
+
+<!-- FIXME -->
```

Comment:
Similarly to network/syslog sources, the $SOURCEIP, $SOURCEPORT macros are always set.

When specifying this option, the $SOURCEIP and $SOURCEPORT  macros will be set according to the defined header, the PEERPORT, PEERIP will contain the address of the proxy.

```
webhook(port(8080) proxy-header("x-forwarded-for"));

$ curl  -H "X-Forwarded-FOR: 1.2.3.4" -X POST --data "{}" http://127.0.0.1:8080/
```

By default, we assume there is no proxy. We only trust the header that is specified in this option, if multiple headers exist in the request with the specified name, the last one is used.